default/hooks/generate-skills-ref.py

```
@@ -294,14 +294,60 @@ def scan_all_plugins(repo_root: Path, plugins: List[str]) -> List[Skill]:
     return aggregated
 
 
+def _max_version_dir(version_dirs: List[Path]) -> Path:
+    """Pick the highest-version directory, comparing version components numerically.
+
+    Plugin caches name dirs by version (``1.39.0``); lexical sorting misorders
+    them (``1.4.0`` > ``1.39.0`` as strings), so compare integer tuples.
+    """
+
+    def key(path: Path):
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `key`

(ANN202)